src/utils/static-eval.js

```diff
@@ -250,12 +250,15 @@ const visitors = {
   },
   MemberExpression (node, walk) {
     const obj = walk(node.object);
-    // do not allow access to methods on Function 
+    // do not allow access to methods on Function
     if (!obj || 'test' in obj || typeof obj.value === 'function')
       return;
     if (node.property.type === 'Identifier') {
       if (typeof obj.value === 'object' && obj.value !== null) {
-        if (node.property.name in obj.value) {
+        if (node.computed) {
+          return { value: WILDCARD, wildcards: [node] };
+        }
```

Comment:
As mentioned above, our goal here should be to get `{}[chunkId]` to evaluate to `undefined`.

Note also that the `WILDCARD` assumption assumes the result to be a string (`WILDCARD` is a string value, used in string concatentation to expand into the globs).

If we were analyzing a member expression that didn't result in a string value, then this would be wrong.

Ideally as mentioned above we could have some `UNKNOWN_WITH_SIDE_EFFECT` and `UNKNOWN_PURE` analysis going on here, but since we don't it is probably best to explicitly handle the `obj[unknownIdentifier]` case directly here for an empty object, along with proper support for computed properties while we are about it.

Something like the following should work - 

```js
if (node.computed) {
  const computedProp = walk(node.property);
  // proper computed property support, because we can and we're here anyway
  if (computedProp && computedProp.value) {
    const val = obj.value[computedProp.value];
    if (val === UNKNOWN) return;
    return { value: v };
  }
  // empty object case for this test
  if (!obj.value[UNKNOWN] && Object.keys(obj).length === 0) {
    return { value: undefined };
  }
}
```

Reply:
I realise that the ideas of pure and not pure are actually ncc-specific, since ncc used to substitute into the expressions (requiring that this analysis is pure and substitutable). So lifting that constraint may mean we can make more analysis optimizations here that ncc can't. Given that you can also ignore my `UNKNOWN_PURE` and `UNKNOWN_WITH_SIDE_EFFECT` comments :)

Reply:
Thanks, I wasn't aware that `{ value: undefined }` had similar semantics to wildcard! Maybe you can explain the 4 to me on Slack one day (wildcard, unknown, { value: undefined }, undefined).

Reply:
`{ value: undefined }` is exactly an undefined value in the analysis, because `{}.x === undefined` in JS is what we want. `UNKNOWN` is mainly used for input values into the analysis boundary through the variables and functions. Otherwise all analysis values are wrapped up in `{ value: ... }` the `undefined` means no analysis available, but note that many cases that return `undefined` can probably be upgraded to return a new `UNKNOWN` analysis value since we are no longer tracking side effects. That is it may make sense to extend `UNKNOWN` from beyond just the analysis inputs. But keeping an eye on performance would be important too as these are super hot paths.